### ISSUE TITLE
使用 XRay 的 Unix domain socket 功能，生成的文件似乎有权限问题。

### ISSUE THREAD

**Issue by mclovin-2k** [XTLS/Xray-core#554]
`    "inbounds": [
        {
            "listen": "/dev/shm/xray_uds_vless_grpc.socket",
            "protocol": "vless",
            ..........
        }
     ]
`

在 inbound 的 listen 属性中，启用 Unix domain socket 之后。生成的对应文件如图。

![QQ截图20210510101846](https://user-images.githubusercontent.com/5440567/117598050-7eb6cb00-b179-11eb-99b9-478136ef68c2.png)

上面4个文件，是 NGINX 的 Unix domain socket 文件，可以看到其权限为所有人可读可写。
后面2个文件是 XRAY 的 Unix domain socket 文件，其权限为所有者可读可写。
这样会导致 NGINX 无法通过 /dev/shm/xray_uds_vless_grpc.socket 和 XRAY 通信。如图：
![QQ截图20210510135015](https://user-images.githubusercontent.com/5440567/117611790-b08a5a80-b196-11eb-90b4-962f712ebf9b.png)

使用 chmod 将 /dev/shm/xray_uds_vless_grpc.socket 权限改为所有人可读可写后，错误消失，通信正常。

**Comment by rurirei** [comment]
you should NOT use root user.

**Comment by changyp6** [comment]
```
 transport/internet/system_listener.go | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/transport/internet/system_listener.go b/transport/internet/system_listener.go
index eb753d6..809f338 100644
--- a/transport/internet/system_listener.go
+++ b/transport/internet/system_listener.go
@@ -4,6 +4,7 @@ import (
        "context"
        "runtime"
        "syscall"
+       "os"
 
        "github.com/pires/go-proxyproto"
        "github.com/xtls/xray-core/common/net"
@@ -45,6 +46,7 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
        var l net.Listener
        var err error
        var network, address string
+       var ds bool
        switch addr := addr.(type) {
        case *net.TCPAddr:
                network = addr.Network()
@@ -72,6 +74,7 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
                                return nil, err
                        }
                        ctx = context.WithValue(ctx, address, locker)
+                       ds = true
                }
        }
 
@@ -80,6 +83,11 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
                policyFunc := func(upstream net.Addr) (proxyproto.Policy, error) { return proxyproto.REQUIRE, nil }
                l = &proxyproto.Listener{Listener: l, Policy: policyFunc}
        }
+       if err == nil && ds {
+               if e := os.Chmod(address, 0777); e != nil {
+                       newError("failed to set file mode of ", address, "to 0777").Base(e).AtWarning().WriteToLog()
+               }
+       }
        return l, err
 }

```

You may apply this patch.
I've found this issue half a year ago and I made this patch.

**Comment by mclovin-2k** [comment]
> you should NOT use root user.

to 

> ```
>  transport/internet/system_listener.go | 8 ++++++++
>  1 file changed, 8 insertions(+)
> 
> diff --git a/transport/internet/system_listener.go b/transport/internet/system_listener.go
> index eb753d6..809f338 100644
> --- a/transport/internet/system_listener.go
> +++ b/transport/internet/system_listener.go
> @@ -4,6 +4,7 @@ import (
>         "context"
>         "runtime"
>         "syscall"
> +       "os"
>  
>         "github.com/pires/go-proxyproto"
>         "github.com/xtls/xray-core/common/net"
> @@ -45,6 +46,7 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
>         var l net.Listener
>         var err error
>         var network, address string
> +       var ds bool
>         switch addr := addr.(type) {
>         case *net.TCPAddr:
>                 network = addr.Network()
> @@ -72,6 +74,7 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
>                                 return nil, err
>                         }
>                         ctx = context.WithValue(ctx, address, locker)
> +                       ds = true
>                 }
>         }
>  
> @@ -80,6 +83,11 @@ func (dl *DefaultListener) Listen(ctx context.Context, addr net.Addr, sockopt *S
>                 policyFunc := func(upstream net.Addr) (proxyproto.Policy, error) { return proxyproto.REQUIRE, nil }
>                 l = &proxyproto.Listener{Listener: l, Policy: policyFunc}
>         }
> +       if err == nil && ds {
> +               if e := os.Chmod(address, 0777); e != nil {
> +                       newError("failed to set file mode of ", address, "to 0777").Base(e).AtWarning().WriteToLog()
> +               }
> +       }
>         return l, err
>  }
> ```
> 
> @mclovin-2k
> You may apply this patch.
> I've found this issue half a year ago and I made this patch.
> You can also change `0777` to `0666`, if you don't want the X bit to be set.

Could you make a PL? That would help us all.

**Comment by mclovin-2k** [comment]
> you should NOT use root user.

That has nothing to do with this issue.
This permission issue will happen, as long as xray and nginx use different user.

**Comment by lelemka0** [comment]
#342

**Comment by yuhan6665** [comment]
Fixed in v1.5.4
